### PR TITLE
Add panel hiding sample for Dock Manager

### PR DESCRIPTION
**Pull request by zhosti**
Closes https://github.com/IgniteUI/igniteui-wc-examples/issues/48
hide-panes

### CODE REVIEW

**Review by gedinakova — APPROVED** [review]

File: samples/layouts/dock-manager/hiding-panes/index.html [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  <body>
+    <div id="root">
+
+<div class="igContainer">
+    <div class="igOptions-horizontal">
+      <label htmlFor="hideOnCloseCheckbox" class="igOptions-label">Hide on close</label>
+      <input id="hideOnCloseCheckbox" type="checkbox" class="igOptions-checkbox"/>
+
+      <label htmlFor="panes" class="igOptions-label">Hidden Panes:</label>
+      <select name="panes" id="panes" class="igOptions-input-text"></select>
```
Comment (DiyanDimitrov) [inline samples/layouts/dock-manager/hiding-panes/index.html:23]: Could we try to better align these two?

**Review by DiyanDimitrov — CHANGES_REQUESTED** [review]

File: samples/layouts/dock-manager/hiding-panes/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+                }
+                ev.preventDefault();
+            }
+        });
+    }
+
+    private setHiddenPane(pane: IgcContentPane) {
+        let option = document.createElement("option");
+        option.textContent = pane.header;
+        option.value = pane.id!;
```
[the remark this answers is not in the document]
Comment (zhosti) [reply samples/layouts/dock-manager/hiding-panes/src/index.ts:157]: Actually no. I was using it for already deleted functionality.

File: samples/layouts/dock-manager/hiding-panes/src/DockManagerStyles.css [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+.dockManagerContent {
+    padding: 0.5rem;
+    height: calc(100% - 1rem);
+    width: calc(100% - 1rem);
+    display: flex;
+    flex-direction: column;
+    /* background: orange; */
+}
+
+.dockManagerFull {
```
[the remark this answers is not in the document]
Comment (zhosti) [reply samples/layouts/dock-manager/hiding-panes/src/DockManagerStyles.css:11]: No. I'll remove them.

File: samples/layouts/dock-manager/hiding-panes/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,184 @@
+import "./DockManagerStyles.css";
+import { defineCustomElements } from "igniteui-dockmanager/loader";
+import { IgcContentPane, IgcDockManagerLayout, IgcDockManagerPaneType, IgcPaneCloseEventArgs } from "igniteui-dockmanager";
+import { IgcSplitPaneOrientation } from "igniteui-dockmanager";
+import { IgcDockManagerComponent } from "igniteui-dockmanager";
```
Comment (DiyanDimitrov) [inline samples/layouts/dock-manager/hiding-panes/src/index.ts:5]: Combine the imports from the "igniteui-dockmanager" module.

File: samples/layouts/dock-manager/hiding-panes/src/index.ts [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+        this.dockManager.layout = { ...this.layout };
+
+        this.handlePaneClose();
+        this.showAllPanes();
+        document.getElementById("showPane")?.addEventListener("click", () => {
+            this.showPane();
+        });
+    }
+
+    public handlePaneClose() {
```
Comment (DiyanDimitrov) [inline samples/layouts/dock-manager/hiding-panes/src/index.ts:137]:
```suggestion
    private handlePaneClose() {
```

**Review by DiyanDimitrov — APPROVED** [review]

**Review by gedinakova — APPROVED** [review]

**Comment by DiyanDimitrov** [comment]
@HUSSAR-mtrela the PR is ready for merging. Could you please review it and merge it if it is ok?

**Review by HUSSAR-mtrela — APPROVED** [review]

**Pull request merged**